plugins/browser-plugin-media-tracking/src/helperFunctions.ts

```diff
@@ -0,0 +1,71 @@
+import { DocumentEvent, TextTrackEvent } from './wgEvents';
+
+export function timeRangesToObjectArray(t: TimeRanges): { start: number; end: number }[] {
+  let out = [];
+  for (let i = 0; i < t.length; i++) {
+    out.push({ start: t.start(i), end: t.end(i) });
+  }
+  return out;
+}
+
+interface TextTrackObject {
+  label: string;
+  language: string;
+  kind: string;
+  mode: string;
+}
+
+export function textTrackListToJson(textTrackList: TextTrackList): TextTrackObject[] {
+  let out: TextTrackObject[] = [];
+  for (let o of Object.keys(textTrackList)) {
+    let i = parseInt(o);
+    out.push({
+      label: textTrackList[i].label,
+      language: textTrackList[i].language,
+      kind: textTrackList[i].kind,
+      mode: textTrackList[i].mode,
+    });
+  }
+  return out;
+}
+
+export function isTypeTextTrackEvent(e: string): boolean {
+  let fields: string[] = Object.values(TextTrackEvent);
+  return fields.includes(e);
+}
+
+export function isTypeDocumentEvent(e: string): boolean {
+  let fields: string[] = Object.values(DocumentEvent);
+  return fields.includes(e);
+}
+
+function getDuplicatesInArray(arr: number[]) {
+  return arr.filter((el, i) => arr.indexOf(el) !== i);
+}
+
+export function checkPercentBoundryArrayIsValid(percentBoundries: number[]): void {
```

Comment:
You could proably use an `is` here. Sort of like this: https://github.com/snowplow/snowplow-javascript-tracker/blob/c2e3ccb94f9ff5fad7ea295d36bffa0c5cd2f819/libraries/browser-tracker-core/src/tracker/out_queue.ts